### ISSUE TITLE
Accessibility issues

### ISSUE THREAD

**Issue by Thatoo** [nextcloud/forms#2858]
**Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!**

Nextcloud Form test with screen reader :

- checkboxes: a bug on the Nextcloud side for checking a box, SPACE SELECTED the voice says it's checked, which isn't the case, a second press on space actually checks the box when the voice only says SPACE.

In the checkboxes, everything needs to be reworked: the voices say completely contradictory things.

- Radio buttons work correctly

- Drop-down menu, at first it leaves the form, then reads the options without recalling the context or the question. Does not read the title of the form drop-down menu, but does read the choices. Reads list items, but doesn't mention which one is selected.

- Can't set a date

- Send form: Dynamic content change, but no notification!

Conclusion: Can't NextcloudForm be used for accessibility? No! Not recommended.

**Expected behavior**
It should be possible to fill a Nextcloud Form using a screen reader independently.

**Nextcloud (please complete the following information):**

- Nextcloud-Version: 31.0.7
- Forms-Version: 5.1.2

**Comment by Chartman123** [comment]
@susnux could this be caused by a bug in the vue lib? I did some debugging and when I used the keyboard to select a checkbox, our `onChange` method always receives the previous value of the selected checkbox upon the first keypress. Only the second keypress of the space bar has the correct values. with the mouse everythings works as expected and the updated values are used in the first call of `onChange`

**Comment by susnux** [comment]
> checkboxes: a bug on the Nextcloud side for checking a box, SPACE SELECTED the voice says it's checked, which isn't the case, a second press on space actually checks the box when the voice only says SPACE.

This sounds like its caused by forms itself

> Drop-down menu

This seems like a problem with NcSelect in general

> Can set a date

I do not understand this issue?

> Send form: Dynamic content change, but no notification!

Which content changes here? I guess something needs a aria live region.

**Comment by pringelmann** [comment]
The date can indeed not be set using keyboard navigation. Attempting to enter the date picker using `Tab` immediately closes it. This will be fixed by https://github.com/nextcloud/forms/pull/2972

**Comment by Chartman123** [comment]
All sub-issues resolved
